### ISSUE TITLE
subkey revocation signature verification fails

### ISSUE THREAD

**Issue by anarcat** [SecurityInnovation/PGPy#226]
One of my authentication subkeys was revoked recently, according to GnuPG:

```
The following key was revoked on 2017-01-16 by RSA key 792152527B75921E Antoine Beaupré <anarcat@anarc.at>
sub  rsa2048/46DC033CAFD0FDF8
     created: 2012-07-24  revoked: 2017-01-16  usage: A   
```

So I should be able to verify this with PGPy:

```python
In [56]: k, _ = pgpy.PGPKey.from_file('anarcat.asc')
In [57]: k.verify(k, k.subkeys['46DC033CAFD0FDF8']._signatures[1])
Out[57]: <SignatureVerification(False)>   
```

Ooops. Maybe it's because i didn't use the right subject:

```python
In [58]: k.verify(k.subkeys['46DC033CAFD0FDF8'], k.subkeys['46DC033CAFD0FDF8']._signatures[1]) 
Out[58]: <SignatureVerification(False)> 
```

Nope. What happened here? Some more details about the signature found:

```python
In [67]: k.subkeys['46DC033CAFD0FDF8']._signatures[1].signer == k.fingerprint.keyid
Out[67]: True
In [68]: k.subkeys['46DC033CAFD0FDF8']._signatures[1].type
Out[68]: <SignatureType.SubkeyRevocation: 40>
```

So I was told this might be a bug by @Commod0re so I'm blindly documenting it here. :)

**Comment by Commod0re** [comment]
thanks, I'm pretty sure this is a bug. I will try to fix this tonight and it'll be included in v0.4.4, which I intend to release as soon as this and the UserID encoding issue are both fixed

**Comment by Commod0re** [comment]
finally tracked this down, subkey revocation signatures need to have the primary key hashed as well.
